main/com.tunein/index.js

```diff
@@ -0,0 +1,158 @@
+// -*- mode: js; indent-tabs-mode: nil; js-basic-offset: 4 -*-
+//
+// Copyright 2021 The Board of Trustees of the Leland Stanford Junior University
+//
+// Redistribution and use in source and binary forms, with or
+// without modification, are permitted provided that the following
+// conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+// 2. Redistributions in binary form must reproduce the above
+//    copyright notice, this list of conditions and the following
+//    disclaimer in the documentation and/or other materials
+//    provided with the distribution.
+// 3. Neither the name of the copyright holder nor the names of its
+//    contributors may be used to endorse or promote products derived
+//    from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+// FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+// COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+// INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+// (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+// HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+// OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Author: Jake Wu <jmhw0123@gmail.com>
+"use strict";
+
+const Tp = require("thingpedia");
+const Http_Status = require('http-status-codes');
+
+const BASE_URL = 'http://opml.radiotime.com';
+const RENDER = '&render=json';
+const CONTENT_TYPE = 'application/x-www-form-urlencoded';
+const QUERY_PARAM = {
+    search: '/Search.ashx',
+    browse: '/Browse.ashx'
+};
+const BROWSE_ITEMS = {
+    trending: 'trending',
+    local: 'local'
+};
+
+function throw_http_error(error_code) {
+    error_code = (typeof error_code == "number") ? error_code : Number(error_code);
+    switch (error_code) {
+        case 400:
+            throw new Error('bad_request')
```

Comment:
The agent looks for the error code in the `code` property of the error object, not the message (which is for logging/debugging), so you need a custom subclass of Error or set the error code on the Error object before throwing.